src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -856,6 +856,9 @@ NESTED_ENTRY JIT_MonTryEnter_InlineGetThread, _TEXT
         mov                 [rsp + 8h], rcx
         .savereg            rdx, 10h
         mov                 [rsp + 10h], rdx
+	
+        push_nonvol_reg     r12 ; rcx now at [rsp + 10h]
+		
```

Comment:
I do not think you need the extra register. The timeout is stored at [rsp + 10h] already, so the two places that need to check it can just do "cmp dword ptr [rsp + 10h], 0"